core/src/execution/datafusion/expressions/negative.rs

```diff
@@ -0,0 +1,397 @@
+        self.arg.data_type(input_schema)
+    }
+
+    fn nullable(&self, input_schema: &Schema) -> Result<bool> {
+        self.arg.nullable(input_schema)
+    }
+
+    fn evaluate(&self, batch: &RecordBatch) -> Result<ColumnarValue> {
+        let arg = self.arg.evaluate(batch)?;        
+        match arg {
+            ColumnarValue::Array(array) => {
+                if self.eval_mode == EvalMode::Ansi {
+                    match array.data_type() {
+                        DataType::Int8 => {
+                            let int_array = array.as_any().downcast_ref::<arrow::array::Int8Array>().unwrap();
+                            for i in 0..int_array.len() {
+                                if int_array.value(i) <= i8::MIN || int_array.value(i) >= i8::MAX {
+                                    return Err(CometError::ArithmeticOverflow{
+                                        from_type: "integer".to_string(),
+                                    }.into());
+                                }
+                            }
+                        }
+                        DataType::Int16 => {
+                            let int_array = array.as_any().downcast_ref::<arrow::array::Int16Array>().unwrap();
+                            for i in 0..int_array.len() {
+                                if int_array.value(i) <= i16::MIN || int_array.value(i) >= i16::MAX {
+                                    return Err(CometError::ArithmeticOverflow{
+                                        from_type: "integer".to_string(),
+                                    }.into());
+                                }
+                            }
+                        }
+                        DataType::Int32 => {
+                            let int_array = array.as_any().downcast_ref::<arrow::array::Int32Array>().unwrap();
+                            for i in 0..int_array.len() {
+                                if int_array.value(i) <= i32::MIN || int_array.value(i) >= i32::MAX {
+                                    return Err(CometError::ArithmeticOverflow{
+                                        from_type: "integer".to_string(),
+                                    }.into());
+                                }
+                            }
+                        }
+                        DataType::Int64 => {
+                            let int_array = array.as_any().downcast_ref::<arrow::array::Int64Array>().unwrap();
+                            for i in 0..int_array.len() {
+                                if int_array.value(i) <= i64::MIN || int_array.value(i) >= i64::MAX {
+                                    return Err(CometError::ArithmeticOverflow{
+                                        from_type: "integer".to_string(),
+                                    }.into());
+                                }
+                            }
+                        }
+                        DataType::UInt8 => {
```

Comment:
Do we ever hit the path for Uint types? Java only has signed integers, so was surprised to see that we need to handle UInt.

Reply:
Hmm... you are right. I added these checks in order to handle all the types present in `ScalarValue`. But it seems like we might not need unsigned types.